contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -1005,6 +1005,47 @@ fooRef:
 
 Although not always necessary to help a controller identify a resource type, “group” is included to avoid ambiguity when the resource exists in multiple groups. It also provides clarity to end users and enables copy-pasting of a reference without the referenced type changing due to a different controller handling the reference.
 
+##### Kind vs. Resource
+
+A common point of confusion in object references is whether to construct
+references with a "kind" or "resource" field. Historically most object
+references in Kubernetes have used "kind". This is not as precise as "resource".
+Although each combination of "group" and "resource" must be unique within
+Kubernetes, the same is not always true for "group" and "kind". It is possible
+for multiple resources to make use of the same "kind".
+
+Typically all objects in Kubernetes have a canonical primary resource - such as
+“pods” representing the way to create and delete resources of the “Pod” schema.
+While it is possible a resource schema cannot be directly created, such as a
+“Scale” object which is only used within the “scale” subresource of a number of
+workloads, most object references address the primary resource via its schema
+("kind").
+
+If an object reference may be used to reference resources of arbitrary types and
+the mapping between kind and resource could be ambiguous, "resource" should be
+used in the object reference.
+
+If implementations of an object reference will always have a clear way to map
+kinds to resources, it is acceptable to use "kind" in the object reference.
+
+The Ingress API provides a good example of where "kind" is appropriate for an
```

Comment:
I think "appropriate" may be too strong of a word. "acceptable" may be better. 

"generic reference" to me is synonymous with "arbitrary reference", which this PR above states that "resource" should be used.

I think a more succinct explanation could be helpful here. This is acceptable because:

- there is a hard-coded mapping of kind to resource.
- the intention is never to support a completely arbitrary set of resources.
- older versions of this API also use kind.

Reply:
Updated, thanks!